process/block/preprocess/transactions.go

```diff
@@ -1334,6 +1349,60 @@ func SortTransactionsBySenderAndNonce(transactions []*txcache.WrappedTransaction
 	sort.Slice(transactions, sorter)
 }
 
+// preFilterTransactions filters the transactions prioritising the move balance operations
+func preFilterTransactions(
+	transactions []*txcache.WrappedTransaction,
+	gasBandwidth uint64,
+	gasHandler process.GasHandler,
+) []*txcache.WrappedTransaction {
+	selectedTxs := make([]*txcache.WrappedTransaction, 0, len(transactions))
+	skippedTxs := make([]*txcache.WrappedTransaction, 0)
+	skippedAddresses := make(map[string]struct{})
+
+	skipped := 0
+	gasEstimation := uint64(0)
+	for i, tx := range transactions {
+		// prioritize move balance operations
+		// and don't care about gas cost for them
+		if len(tx.Tx.GetData()) > 0 {
+			skippedTxs = append(skippedTxs, transactions[i])
+			skippedAddresses[string(tx.Tx.GetSndAddr())] = struct{}{}
+			skipped++
+			continue
+		}
+
+		if shouldSkipTransactionIfMarkedAddress(tx, skippedAddresses) {
+			skippedTxs = append(skippedTxs, transactions[i])
+			continue
+		}
+
+		selectedTxs = append(selectedTxs, transactions[i])
+		gasEstimation += tx.Tx.GetGasLimit()
+	}
+
+	for i, tx := range skippedTxs {
```

Comment:
this is optional. We could have just returned the `selectedTxs` slice appended with the `skippedTxs` in order to spare CPU processing time